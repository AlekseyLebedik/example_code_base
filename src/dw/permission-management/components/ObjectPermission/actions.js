import * as actionTypes from './actionTypes';

export const fetchContentType = id => ({
  type: actionTypes.CONTENT_TYPE_FETCH,
  id,
});

export const fetchRelatedPermissions = (contentTypeId, id) => ({
  type: actionTypes.FETCH_RELATED_PERMISSONS,
  id,
  contentTypeId,
});
export const fetchRelatedPermissionsSuccess = data => ({
  type: actionTypes.FETCH_RELATED_PERMISSONS_SUCCESS,
  data,
});
export const fetchContentTypeSuccess = (id, data) => ({
  type: actionTypes.CONTENT_TYPE_FETCH_SUCCESS,
  id,
  data,
});

export const fetchContentTypeFail = (id, error) => ({
  type: actionTypes.CONTENT_TYPE_FETCH_FAIL,
  id,
  error,
});

export const fetchObjectPermissions = (entityType, ctypeId, objectId) => ({
  type: actionTypes.OBJECT_PERMISSIONS_FETCH,
  entityType,
  ctypeId,
  objectId,
});

export const fetchObjectPermissionsSuccess = (
  entityType,
  ctypeId,
  objectId,
  data
) => ({
  type: actionTypes.OBJECT_PERMISSIONS_FETCH_SUCCESS,
  entityType,
  ctypeId,
  objectId,
  data,
});

export const fetchObjectPermissionsFail = (
  entityType,
  ctypeId,
  objectId,
  error
) => ({
  type: actionTypes.OBJECT_PERMISSIONS_FETCH_FAIL,
  entityType,
  ctypeId,
  objectId,
  error,
});

export const fetchCompanies = () => ({
  type: actionTypes.COMPANIES_FETCH,
});

export const fetchCompaniesSuccess = (data, append = false) => ({
  type: actionTypes.COMPANIES_FETCH_SUCCESS,
  data,
  append,
});

export const fetchCompaniesFail = error => ({
  type: actionTypes.COMPANIES_FETCH_FAIL,
  error,
});

export const fetchCompanyGroups = () => ({
  type: actionTypes.COMPANY_GROUPS_FETCH,
});

export const fetchCompanyGroupsSuccess = data => ({
  type: actionTypes.COMPANY_GROUPS_FETCH_SUCCESS,
  data,
});

export const fetchCompanyGroupsFail = error => ({
  type: actionTypes.COMPANY_GROUPS_FETCH_FAIL,
  error,
});

export const grantPermissions = (
  entityType,
  ctypeId,
  objectId,
  permissions
) => ({
  type: actionTypes.GRANT_PERMISSIONS,
  entityType,
  ctypeId,
  objectId,
  permissions,
});

export const grantPermissionsSuccess = (
  entityType,
  ctypeId,
  objectId,
  permissions
) => ({
  type: actionTypes.GRANT_PERMISSIONS_SUCCESS,
  entityType,
  ctypeId,
  objectId,
  permissions,
});

export const grantPermissionsFail = (
  entityType,
  ctypeId,
  objectId,
  permissions,
  error
) => ({
  type: actionTypes.GRANT_PERMISSIONS_FAIL,
  entityType,
  ctypeId,
  objectId,
  permissions,
  error,
});

export const revokePermissions = (
  entityType,
  ctypeId,
  objectId,
  permissions
) => ({
  type: actionTypes.REVOKE_PERMISSIONS,
  entityType,
  ctypeId,
  objectId,
  permissions,
});

export const revokePermissionsSuccess = (
  entityType,
  ctypeId,
  objectId,
  permissions
) => ({
  type: actionTypes.REVOKE_PERMISSIONS_SUCCESS,
  entityType,
  ctypeId,
  objectId,
  permissions,
});

export const revokePermissionsFail = (
  entityType,
  ctypeId,
  objectId,
  permissions,
  error
) => ({
  type: actionTypes.REVOKE_PERMISSIONS_FAIL,
  entityType,
  ctypeId,
  objectId,
  permissions,
  error,
});
