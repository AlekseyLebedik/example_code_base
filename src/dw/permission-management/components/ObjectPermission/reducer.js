import * as actionTypes from './actionTypes';

export const INITIAL_STATE = {
  contentTypePermissions: {},
  objectPermissions: {},
  relatedPermissions: [],
  companies: [],
  companyGroups: [],
  Components: {
    ObjectPermissionManager: {
      initializationError: null,
    },
  },
};

export default function reducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case actionTypes.CONTENT_TYPE_FETCH_SUCCESS:
      return {
        ...state,
        contentTypePermissions: {
          ...state.contentTypePermissions,
          [action.id]: action.data,
        },
      };
    case actionTypes.COMPANIES_FETCH_SUCCESS:
      return {
        ...state,
        companies: action.append
          ? [...state.companies, ...action.data]
          : action.data,
      };
    case actionTypes.COMPANY_GROUPS_FETCH_SUCCESS:
      return {
        ...state,
        companyGroups: action.data,
      };
    case actionTypes.OBJECT_PERMISSIONS_FETCH_SUCCESS:
      return {
        ...state,
        objectPermissions: {
          ...state.objectPermissions,
          [action.ctypeId]: {
            ...state.objectPermissions[action.ctypeId],
            [action.entityType]: action.data,
          },
        },
      };
    case actionTypes.FETCH_RELATED_PERMISSONS_SUCCESS:
      return {
        ...state,
        relatedPermissions: [...action.data.data],
      };
    case actionTypes.COMPANIES_FETCH_FAIL:
    case actionTypes.COMPANY_GROUPS_FETCH_FAIL:
    case actionTypes.CONTENT_TYPE_FETCH_FAIL:
    case actionTypes.OBJECT_PERMISSIONS_FETCH_FAIL:
      return {
        ...state,
        Components: {
          ...state.Components,
          ObjectPermissionManager: { initializationError: action.error },
        },
      };
    default:
      return state;
  }
}
