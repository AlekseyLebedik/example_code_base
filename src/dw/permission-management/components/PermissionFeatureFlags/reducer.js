import { createFetchReducer } from '@demonware/devzone-core/helpers/reducers';
import * as AT from './actionTypes';

const fetchUsersListReducer = createFetchReducer(AT.FEATURE_FLAG_PERMISSIONS);

export default fetchUsersListReducer;
