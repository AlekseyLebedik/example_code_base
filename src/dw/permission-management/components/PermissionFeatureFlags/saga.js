import * as API from 'dw/permission-management/services/explainerflags';
import { getSaga } from '@demonware/devzone-core/helpers/sagas';
import * as AT from './actionTypes';

const fetchPermFeatureFlagsSaga = getSaga(AT.FEATURE_FLAG_PERMISSIONS, params =>
  API.getFeatureFlag(params)
);

export default fetchPermFeatureFlagsSaga;
