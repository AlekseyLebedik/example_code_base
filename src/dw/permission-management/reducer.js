import { combineReducers } from 'redux';
import { reducer as formReducer } from 'redux-form';

import errors from '@demonware/devzone-core/helpers/errors';
import criticalErrorReducer from 'dw/core/components/CriticalError/reducer';

import UserReplica from 'dw/core/replicas/user';
import PermissionsReplica from 'dw/core/replicas/permissions';
import SwitchesReplica from 'dw/core/replicas/switches';
import ContentTypeReplica from 'dw/core/replicas/contentType';
import modalHandlersReducer from 'dw/core/components/ModalHandlers/reducer';

import objectPermissionManagerReducer from './components/ObjectPermission/reducer';
import permissionFeatureFlagsReducer from './components/PermissionFeatureFlags/reducer';
import scenesReducer from './scenes/reducer';

const reducer = combineReducers({
  Scenes: scenesReducer,
  Core: combineReducers({
    CriticalError: criticalErrorReducer,
    ModalHandlers: modalHandlersReducer,
    ObjectPermissionManager: objectPermissionManagerReducer,
    PermissionFeatureFlags: permissionFeatureFlagsReducer,
  }),
  errors,
  form: formReducer,
  user: UserReplica.reducer,
  permissions: PermissionsReplica.reducer,
  switches: SwitchesReplica.reducer,
  contentType: ContentTypeReplica.reducer,
});

export default reducer;
