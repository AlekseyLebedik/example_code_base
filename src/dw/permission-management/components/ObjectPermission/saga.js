import { call, put, takeEvery, takeLatest } from 'redux-saga/effects';
import * as api from '@demonware/devzone-core/services/permissions';
// mocks
import { nonCriticalError } from '@demonware/devzone-core/helpers/errors';
import { show } from '@demonware/devzone-core/components/GlobalSnackBar/actions';

import * as actions from './actions';
import * as actionTypes from './actionTypes';

function* fetchContentTypePermissions(action) {
  const { id } = action;
  try {
    const {
      data: { data },
    } = yield call(api.getContentTypePermissions, id);
    yield put(actions.fetchContentTypeSuccess(id, data));
  } catch (e) {
    yield put(actions.fetchContentTypeFail(id, e));
  }
}

function* fetchCompanies(params) {
  try {
    const { next } = params;
    const { data } = yield call(api.getCompanies, next);
    yield put(actions.fetchCompaniesSuccess(data.data, !!next));
    if (!data.next) return;
    yield* fetchCompanies({ ...params, next: data.next });
  } catch (e) {
    yield put(actions.fetchCompaniesFail(e));
  }
}

function* fetchObjectPermissions(action) {
  const { entityType, ctypeId, objectId } = action;
  const apiFn = {
    user: api.getUserObjectPermissions,
    company: api.getCompanyObjectPermission,
    group: api.getCompanyGroupObjectPermissions,
  }[entityType];

  try {
    if (apiFn === undefined)
      throw new Error(`No function for entityType ${entityType}`);

    const {
      data: { data },
    } = yield call(apiFn, ctypeId, objectId);
    yield put(
      actions.fetchObjectPermissionsSuccess(entityType, ctypeId, objectId, data)
    );
  } catch (e) {
    actions.fetchObjectPermissionsFail(entityType, ctypeId, objectId, e);
  }
}

function* fetchCompanyGroups() {
  try {
    const {
      data: { data },
    } = yield call(api.getCompanyGroups);
    yield put(actions.fetchCompanyGroupsSuccess(data));
  } catch (e) {
    yield put(actions.fetchCompanyGroupsFail(e));
  }
}

function* grantPermissions(action) {
  const { entityType, ctypeId, objectId, permissions } = action;
  const apiFn = {
    user: api.addUserObjectPermissions,
    company: api.addCompanyObjectPermissions,
    group: api.addCompanyGroupObjectPermissions,
  }[entityType];

  try {
    if (apiFn === undefined)
      throw new Error(`No function for entityType ${entityType}`);

    if (permissions.length > 0)
      yield call(apiFn, ctypeId, objectId, permissions);

    yield put(
      actions.grantPermissionsSuccess(
        entityType,
        ctypeId,
        objectId,
        permissions
      )
    );
  } catch (e) {
    yield put(
      actions.grantPermissionsFail(
        entityType,
        ctypeId,
        objectId,
        permissions,
        e
      )
    );
  }
}

function* revokePermissions(action) {
  const { entityType, ctypeId, objectId, permissions } = action;
  const apiFn = {
    user: api.deleteUserObjectPermissions,
    company: api.deleteCompanyObjectPermissions,
    group: api.deleteCompanyGroupObjectPermissions,
  }[entityType];

  try {
    if (apiFn === undefined)
      throw new Error(`No function for entityType ${entityType}`);

    if (permissions.length > 0)
      yield call(apiFn, ctypeId, objectId, permissions);

    yield put(
      actions.revokePermissionsSuccess(
        entityType,
        ctypeId,
        objectId,
        permissions
      )
    );
  } catch (e) {
    yield put(
      actions.revokePermissionsFail(
        entityType,
        ctypeId,
        objectId,
        permissions,
        e
      )
    );
  }
}

function* showSuccessMessage() {
  yield put(show(`Permissions set successfully`, 'success'));
}

function* showGrantPermissionFailure({ entityType }) {
  yield put(nonCriticalError(`Error granting ${entityType} permissions`));
}

function* showRevokePermissionFailure({ entityType }) {
  yield put(nonCriticalError(`Error revoking ${entityType} permissions`));
}

function* fetchRelatedObjectPermissions({ contentTypeId, id }) {
  try {
    const { data } = yield call(api.getRelatedPermissions, id, contentTypeId);
    yield put(actions.fetchRelatedPermissionsSuccess(data));
  } catch (e) {
    console.log('error ', e);
  }
}

function* saga() {
  yield takeEvery(actionTypes.CONTENT_TYPE_FETCH, fetchContentTypePermissions);
  yield takeLatest(actionTypes.COMPANIES_FETCH, fetchCompanies);
  yield takeLatest(actionTypes.COMPANY_GROUPS_FETCH, fetchCompanyGroups);
  yield takeEvery(actionTypes.GRANT_PERMISSIONS, grantPermissions);
  yield takeEvery(actionTypes.REVOKE_PERMISSIONS, revokePermissions);
  yield takeEvery(
    actionTypes.GRANT_PERMISSIONS_FAIL,
    showGrantPermissionFailure
  );
  yield takeEvery(
    actionTypes.REVOKE_PERMISSIONS_FAIL,
    showRevokePermissionFailure
  );
  yield takeLatest(actionTypes.GRANT_PERMISSIONS_SUCCESS, showSuccessMessage);

  yield takeLatest(actionTypes.REVOKE_PERMISSIONS_SUCCESS, showSuccessMessage);
  yield takeEvery(actionTypes.OBJECT_PERMISSIONS_FETCH, fetchObjectPermissions);
  yield takeLatest(
    actionTypes.FETCH_RELATED_PERMISSONS,
    fetchRelatedObjectPermissions
  );
}

export default saga;
