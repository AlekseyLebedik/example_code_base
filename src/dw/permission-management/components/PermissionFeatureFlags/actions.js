import { createFetch } from '@demonware/devzone-core/helpers/actions';
import * as AT from './actionTypes';

export const fetchfeaturePermissions = (params = {}) =>
  createFetch(AT.FEATURE_FLAG_PERMISSIONS, null, params, params.nextPageToken);
