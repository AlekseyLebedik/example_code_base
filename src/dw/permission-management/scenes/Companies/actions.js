import { createFetch } from '@demonware/devzone-core/helpers/actions';
import { nonCriticalHTTPError } from '@demonware/devzone-core/helpers/errors';
import {
  CONTENT_TYPES_PREFIX,
  OBJECT_PERMISSIONS_PREFIX,
  CONTENT_TYPES_DETAILS_PREFIX,
  COMPANY_USERS_PREFIX,
} from './constants';

import {
  createEditObjectPermissions,
  createFetchObjectPermissions,
} from '../actionCreators';

export const fetchContentTypes = () => createFetch(CONTENT_TYPES_PREFIX);

export const fetchObjectPermissions = createFetchObjectPermissions(
  OBJECT_PERMISSIONS_PREFIX
);

export const editObjectPermissions = createEditObjectPermissions(
  OBJECT_PERMISSIONS_PREFIX
);

export const fetchContentTypesDetails = (contentTypes, companyIds) => ({
  type: CONTENT_TYPES_DETAILS_PREFIX,
  contentTypes,
  companyIds,
});

export const editCompanyUsers = (companyId, data) => ({
  type: `${COMPANY_USERS_PREFIX}_PUT`,
  companyId,
  data,
});

export const editCompanyUsersFailed = error => dispatch => {
  dispatch(nonCriticalHTTPError(error));
};
