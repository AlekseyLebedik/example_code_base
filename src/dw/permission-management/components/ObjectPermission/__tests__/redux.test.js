import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';

import wait from 'dw/test-utils/wait';
import * as api from '@demonware/devzone-core/services/permissions';

import * as actions from '../actions';
import reducer from '../reducer';
import saga from '../saga';

jest.mock('@demonware/devzone-core/services/permissions');

describe('ObjectPermission', () => {
  describe('data', () => {
    let store;
    beforeEach(() => {
      jest.resetAllMocks();
      const sagaMiddleware = createSagaMiddleware();
      store = createStore(reducer, applyMiddleware(sagaMiddleware));
      sagaMiddleware.run(saga);
    });

    describe('companies', () => {
      it('should store the company information once fetched', async () => {
        api.getCompanies.mockReturnValue(
          Promise.resolve({
            data: {
              data: ['company 1'],
            },
          })
        );

        expect(store.getState()).toMatchSnapshot();
        store.dispatch(actions.fetchCompanies());
        await wait(0);
        expect(store.getState()).toMatchSnapshot();
      });

      it('should store information about all companies', async () => {
        api.getCompanies
          .mockReturnValueOnce(
            Promise.resolve({ data: { data: ['company 1'], next: 'next' } })
          )
          .mockReturnValueOnce(
            Promise.resolve({ data: { data: ['company 2'], next: 'next' } })
          )
          .mockReturnValueOnce(
            Promise.resolve({ data: { data: ['company 3'] } })
          );

        store.dispatch(actions.fetchCompanies());
        await wait(0);
        expect(store.getState()).toMatchSnapshot();
      });

      it('should sotre ObjectPermissionManager.initializationError when it fails to load companies', async () => {
        api.getCompanies.mockReturnValue(Promise.reject(new Error()));
        expect(store.getState()).toMatchSnapshot();
        store.dispatch(actions.fetchCompanies());
        await wait(0);
        expect(store.getState()).toMatchSnapshot();
      });
    });

    describe('company-groups', () => {
      it('should store the company groups info once fetched', async () => {
        api.getCompanyGroups.mockReturnValue(
          Promise.resolve({
            data: {
              data: ['company group 1'],
            },
          })
        );

        expect(store.getState()).toMatchSnapshot();
        store.dispatch(actions.fetchCompanyGroups());
        await wait(0);
        expect(store.getState()).toMatchSnapshot();
      });

      it('should store ObjectPermissionManager.initializationError when it fails to load company groups', async () => {
        api.getCompanyGroups.mockReturnValue(Promise.reject(new Error()));
        expect(store.getState()).toMatchSnapshot();
        store.dispatch(actions.fetchCompanyGroups());
        await wait(0);
        expect(store.getState()).toMatchSnapshot();
      });
    });

    describe('object permissions', () => {
      it('should store the object permissions once fetched', async () => {
        api.getUserObjectPermissions.mockReturnValue(
          Promise.resolve({
            data: {
              data: ['permission 1'],
            },
          })
        );

        expect(store.getState()).toMatchSnapshot();
        store.dispatch(actions.fetchObjectPermissions('user'));
        await wait(0);
        expect(store.getState()).toMatchSnapshot();
      });

      it('should store ObjectPermissionManager.initializationError when it fails to load company groups', async () => {
        api.getUserObjectPermissions.mockReturnValue(
          Promise.reject(new Error())
        );
        expect(store.getState()).toMatchSnapshot();
        store.dispatch(actions.fetchObjectPermissions('user'));
        await wait(0);
        expect(store.getState()).toMatchSnapshot();
      });
    });
  });
});
