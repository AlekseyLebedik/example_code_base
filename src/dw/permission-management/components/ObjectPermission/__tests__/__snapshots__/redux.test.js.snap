// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectPermission data companies should sotre ObjectPermissionManager.initializationError when it fails to load companies 1`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data companies should sotre ObjectPermissionManager.initializationError when it fails to load companies 2`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": [Error],
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data companies should store information about all companies 1`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [
    "company 1",
    "company 2",
    "company 3",
  ],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data companies should store the company information once fetched 1`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data companies should store the company information once fetched 2`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [
    "company 1",
  ],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data company-groups should store ObjectPermissionManager.initializationError when it fails to load company groups 1`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data company-groups should store ObjectPermissionManager.initializationError when it fails to load company groups 2`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": [Error],
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data company-groups should store the company groups info once fetched 1`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data company-groups should store the company groups info once fetched 2`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [
    "company group 1",
  ],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data object permissions should store ObjectPermissionManager.initializationError when it fails to load company groups 1`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data object permissions should store ObjectPermissionManager.initializationError when it fails to load company groups 2`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data object permissions should store the object permissions once fetched 1`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {},
  "relatedPermissions": Array [],
}
`;

exports[`ObjectPermission data object permissions should store the object permissions once fetched 2`] = `
Object {
  "Components": Object {
    "ObjectPermissionManager": Object {
      "initializationError": null,
    },
  },
  "companies": Array [],
  "companyGroups": Array [],
  "contentTypePermissions": Object {},
  "objectPermissions": Object {
    "undefined": Object {
      "user": Array [
        "permission 1",
      ],
    },
  },
  "relatedPermissions": Array [],
}
`;
